frontend/src/lib/StatusUtils.test.tsx

```diff
@@ -130,4 +135,59 @@ describe('StatusUtils', () => {
       expect(checkIfTerminated(NodePhase.FAILED, 'some random error')).toEqual(NodePhase.FAILED);
     });
   });
+
+  describe('parseNodePhase', () => {
+    const DEFAULT_NODE_STATUS = ({
+      phase: 'Succeeded',
+      id: 'file-passing-pipelines-55slt-2894085459',
+      outputs: {
+        artifacts: [
+          ({
+            s3: {
+              key:
+                'artifacts/file-passing-pipelines-55slt/file-passing-pipelines-55slt-2894085459/sum-numbers-output.tgz',
+            },
+          } as unknown) as Artifact,
+        ],
+      },
+    } as unknown) as NodeStatus;
+
+    it('returns node original phase if not successful', () => {
+      expect(
+        parseNodePhase({
+          ...DEFAULT_NODE_STATUS,
+          phase: 'Failed',
+        }),
+      ).toEqual('Failed');
+    });
+
+    it('returns succeeded phase for a normal node', () => {
+      expect(
+        parseNodePhase({
+          ...DEFAULT_NODE_STATUS,
+          phase: 'Succeeded',
+        }),
+      ).toEqual('Succeeded');
+    });
+
+    it('returns cached phase for a cached node', () => {
+      expect(
+        parseNodePhase({
+          ...DEFAULT_NODE_STATUS,
```

Comment:
Are you sure this status is correct?

Reply:
Moved phase explicitly here (the ... syntax means expanding that object, and we override some attributes).